### PR TITLE
TR now will not apply a new cr config if it has https delivery services without certs

### PR DESCRIPTION
**Pull request by trevorackerman**

### CODE REVIEW

File: traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/config/CertificateChecker.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package com.comcast.cdn.traffic_control.traffic_router.core.config;
+
+import com.comcast.cdn.traffic_control.traffic_router.keystore.KeyStoreHelper;
+import org.apache.log4j.Logger;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+public class CertificateChecker {
+	private final static Logger LOGGER = Logger.getLogger(CertificateChecker.class);
+	private final KeyStoreHelper keyStoreHelper = KeyStoreHelper.getInstance();
```
Comment (robert-butts) [inline traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/config/CertificateChecker.java]: This must be mutated by another thread, if it's possible for certificatesAreValid() in ConfigHandler.java:135 to become true, right?

Then, either `KeyStoreHelper.KeyStore` or `KeyStoreHelper` needs declared volatile, right?

File: traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/secure/CertificatesClient.java [hunk 2]
```diff
@@ -36,18 +47,18 @@ public void refreshData() {
 		persistCertificates(certificateDataList);
 	}
 
-	public String fetchRawData() {
+	public int fetchRawData(final StringBuilder stringBuilder) {
```
Comment (robert-butts) [inline traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/secure/CertificatesClient.java]: A comment indicating the return value would be nice here. You can't really tell what the int is without reading the full function.

File: traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/secure/CertificateData.java [hunk 3]
```diff
@@ -27,24 +30,34 @@ public void setCertificate(final Certificate certificate) {
...
+	}
+
+	public void setHostname(final String hostname) {
+		this.hostname = hostname;
+	}
+
 	@SuppressWarnings("PMD")
-	public boolean equals(final Object o) {
+	@Override
+	public boolean equals(Object o) {
```
Comment (robert-butts) [inline traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/secure/CertificateData.java]: Why was this changed to not be final? The `equals` function shouldn't modify it, right?

File: traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/secure/CertificateData.java [hunk 4]
```diff
@@ -27,24 +30,34 @@ public void setCertificate(final Certificate certificate) {
...
 		if (this == o) return true;
 		if (o == null || getClass() != o.getClass()) return false;
 
-		final CertificateData that = (CertificateData) o;
+		CertificateData that = (CertificateData) o;
 
 		if (deliveryservice != null ? !deliveryservice.equals(that.deliveryservice) : that.deliveryservice != null)
 			return false;
-		return certificate != null ? certificate.equals(that.certificate) : that.certificate == null;
+		if (certificate != null ? !certificate.equals(that.certificate) : that.certificate != null) return false;
```
Comment (robert-butts) [inline traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/secure/CertificateData.java]: I had to stare at this for like 45 seconds to figure out what it does in every case. If it were two separate `if`s, it'd take, like, 10% the time to mentally parse.

File: traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/secure/CertificateData.java (hunk 3 above)
Comment (trevorackerman) [inline traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/secure/CertificateData.java]: I updated the class with a field told IntelliJ to regenerate the equals method.
That caused the final keyword to be dropped unintentionally, I usually add it back.

I always use IntelliJ's generation of equals and hashCode, it doesn't read well but it always works and is performant.

File: traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/secure/CertificateData.java (hunk 4 above)
Comment (trevorackerman) [inline traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/secure/CertificateData.java]: see my comment above
